actual/api/bank_sync.py

```diff
@@ -60,7 +60,7 @@ class Balance(BaseModel):
 
     balance_amount: BankSyncAmount = Field(..., alias="balanceAmount")
     balance_type: BalanceType = Field(..., alias="balanceType")
-    reference_date: Optional[str] = Field(..., alias="referenceDate", description="The date of the balance")
+    reference_date: Optional[str] | None = Field(None, alias="referenceDate", description="The date of the balance")
```

Comment:
Here you are trying to type None twice, since the `Optional` already implies `str | None`. You can do:

```suggestion
    reference_date: Optional[str] = Field(None, alias="referenceDate", description="The date of the balance")
```

Tested with:

```python
from actual.api.bank_sync import Balance

Balance.model_validate({"balanceAmount": {"amount": 50, "currency": "EUR"}, "balanceType": "interimAvailable"})
```